weave-js/src/components/PagePanelComponents/Home/Browse3/compare/CompareGridCell.tsx

```diff
@@ -24,6 +25,37 @@ type CompareGridCellProps = {
   rowChangeType: any;
 };
 
+function limitStringLength(str: string, maxLength: number) {
+  if (str.length > maxLength) {
+    return str.slice(0, maxLength) + '...';
+  }
+  return str;
+}
+
+/**
+ * Detects if a string is likely a base64 encoded image.
+ *
+ * @param str - The string to check
+ * @returns true if the string appears to be a base64 image
+ */
+function isBase64Image(str: string): boolean {
+  if (!str || typeof str !== 'string') {
+    return false;
+  }
+
+  // Check for data URL format: data:image/[type];base64,[data]
+  if (str.startsWith('data:image/') && str.includes(';base64,')) {
+    return true;
+  }
+
+  // Check for 'base64:' prefix format
+  if (str.startsWith('base64:')) {
+    return true;
```

Comment:
totally optional: could consider regex.test, should be slightly more performant and we can make both conditions in one